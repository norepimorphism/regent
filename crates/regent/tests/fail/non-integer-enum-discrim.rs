fn main() {
    // Array.
    #[regent::bitwise]
    enum A { _0 = [0] }
    // Boolean.
    #[regent::bitwise]
    enum B { _0 = true }
    // Character.
    #[regent::bitwise]
    enum C { _0 = 'a' }
    // String.
    #[regent::bitwise]
    enum D { _0 = "abc" }
    // Tuple.
    #[regent::bitwise]
    enum E { _0 = () }
}
