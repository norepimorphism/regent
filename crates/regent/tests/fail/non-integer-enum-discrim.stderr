error: variant discriminant must be an integer literal
 --> tests/fail/non-integer-enum-discrim.rs:4:19
  |
4 |     enum A { _0 = [0] }
  |                   ^^^

error: variant discriminant must be an integer literal
 --> tests/fail/non-integer-enum-discrim.rs:7:19
  |
7 |     enum B { _0 = true }
  |                   ^^^^

error: variant discriminant must be an integer literal
  --> tests/fail/non-integer-enum-discrim.rs:10:19
   |
10 |     enum C { _0 = 'a' }
   |                   ^^^

error: variant discriminant must be an integer literal
  --> tests/fail/non-integer-enum-discrim.rs:13:19
   |
13 |     enum D { _0 = "abc" }
   |                   ^^^^^

error: variant discriminant must be an integer literal
  --> tests/fail/non-integer-enum-discrim.rs:16:19
   |
16 |     enum E { _0 = () }
   |                   ^^
